### DISCUSSION TITLE
Is there a way to log consumer id in kong (nginx) access log ?

### DISCUSSION THREAD

**Discussion by adarshk-rapido** [Kong/kong#8709] [Help] [answered]
First and foremost thanks to everyone working on Kong and supporting the community,

We are trying to figure out a weird bug and would like to know if there is a way to log the consumer id in kong access logs. I know while proxying kong add x-consumer-username but that wont be available at access log side right.

**Comment by flrgh** [comment, marked as answer]
hi @adarshk-rapido :wave: 

The request consumer is not available within `access_log` because it's not a NGINX-level variable. Most all of the common logging [plugins](https://docs.konghq.com/hub/) will include the consumer in the json log entry that they emit, so it's probably easiest to use one of these to get the debug info you need.

**Reply by hbagdi** [reply]
Could one update a header in the request and then use the header to log?

**Reply by adarshk-rapido** [reply]
@flrgh thanks for the input let me give the logging plugin a try.

@hbagdi yes we can try to add another unique ID instead, so that can help us debug as well. Thank you
